src/app/static/src/js/components/reportLog/runningReportDetails.vue

```diff
@@ -133,28 +137,44 @@
         },
         methods: {
             getLogs: function () {
-                api.get(`/running/${this.reportKey}/logs/`)
-                    .then(({data}) => {
-                        this.reportLog = data.data
-                        this.error = "";
-                        this.defaultMessage = "";
-                    })
-                    .catch((error) => {
-                        this.error = error;
-                        this.defaultMessage = "An error occurred when fetching logs";
-                    });
+                if (this.reportKey) {
+                    api.get(`/running/${this.reportKey}/logs/`)
+                        .then(({data}) => {
+                            this.reportLog = data.data;
+                            this.error = "";
+                            this.defaultMessage = "";
+
+                            const status = this.reportLog.status;
+                            if (status === "running" || status === "queued") {
+                                if (!this.pollingTimer) {
```

Comment:
If you wanted to be super-defensive and localise all access to `pollingTimer` to the `st*Polling` functions then you could move this check to within `startPolling` itself.